features/mbedtls/platform/inc/mbedtls_config_reader.h

```diff
@@ -0,0 +1,500 @@
+/*
+ * Copyright (c) 2013-2018, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MBEDTLS_CONFIG_READER_H__
+#define __MBEDTLS_CONFIG_READER_H__
+
+#include "platform/inc/platform_mbed.h"
+
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(MBEDTLS_TEST_NULL_ENTROPY) && \
+    !defined(MBEDTLS_ENTROPY_NV_SEED)
+    // for platforms that don't have entropy,none of the ciphersuites will
+    // work, so don't bother
```

Comment:
From a usability point of view, I would prefer if we threw and error here instead of silently disabling the whole of Mbed TLS and then just throwing a linker error if the user's program relies on one of the library functions. What is the motivation for the existing behaviour instead or throwing an error (or at least a warning)?

Reply:
Throwing an error for each of these adds approximately 100 lines of code that need to be maintained.
I will add them, however.